audformat/core/conftest.py

```diff
@@ -0,0 +1,26 @@
+import os
+
+import numpy as np
+import pytest
+
+import audeer
+import audiofile
+
+
+@pytest.fixture(scope='session', autouse=True)
+def prepare_docstring_tests():
+
+    # Prepare WAV files needed in doctests
+    #
+    # audformat.utils.to_filewise_index()
+    audiofile.write('f.wav', np.ones((1, 8000)), 8000)
```

Comment:
Maybe use a temporary directory as in:

https://github.com/audeering/audbackend/blob/dev/audbackend/core/conftest.py#L10

?

Reply:
Then you don't have to care about removing files after `yield`.

Reply:
I tried, see https://github.com/audeering/audformat/pull/370/commits/a69365a2740de6d656c5d5c0f04002cf9078683e, but I don't know why it is not working.

Reply:
Also the example seems slightly easier to me without the `tmp` folder even though it is a much cleaner solution.

Reply:
I found the error, I forgot to indent `yield` as well.

Reply:
Yes, otherwise the audio file is removed before the example is executed :)

Reply:
That is strange, why does it now include the absolute path and not before?

Reply:
Maybe a bug in `to_filewise_index()`?

Reply:
No, it depends what you provide as `output_folder`:

![image](https://user-images.githubusercontent.com/173624/232005160-df14fcc5-f264-4c6a-9a71-40de19c06109.png)

Reply:
But your output folder was relative, wasn't it?

Reply:
No, I had to write `audeer.path(tmp, 'split')` to make sure the resulting files are stored in the tmp folder and automatically cleaned.
Without using the tmp folder, the path is relative.

Reply:
But the current working directory should be set to `tmp` in the docstring example.

Reply:
No, this is not automatically happening.
I can enforce it with `os.chdir(tmp)`, but then other tests fail (even when I change the directory back after `yield`).

Reply:
I will try to change those other failing tests, and see if it makes sense.

Reply:
Currently blocked by https://github.com/audeering/audformat/issues/371

Reply:
#371 is merged and I was now able to use a tmp folder to store any files.

We have to change folders during tests, but it doesn't seem to have any undesired consequences.